### PR TITLE
Add secret detection rules using regexes

### PR DESCRIPTION
**Pull request by minusworld**
Bunch of secret detection regexes. Related to https://github.com/returntocorp/semgrep-rules/issues/731.

### CODE REVIEW

**Review by inkz — APPROVED** [review]
:+1:

File: regex/secrets/detected-artifactory-token.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+rules:
+- id: detected-artifactory-token
+  pattern-regex: |-
+    AKC[a-zA-Z0-9]{10,}
+  languages: [regex]
```
Comment (dlukeomalley) [inline regex/secrets/detected-artifactory-token.yaml:5]: What is the `regex` language?

**Review by dlukeomalley — COMMENTED** [review]
@minusworld I'm thrilled about this PR! Couple questions:

1. Is there any benefit to restricting regex to string literals? Curious if it helps with either accuracy and/or speed (reducing the search space, but increasing the parsing?)
2. Are there any file extensions that can be used to reduce the search space? I noticed while looking at other "generic" rules we haven't made a habit of it, and curious here if it would help  or be possible.

File: regex/secrets/detected-artifactory-token.yaml (hunk 1 above)
Comment (minusworld) [reply regex/secrets/detected-artifactory-token.yaml:5]: `regex` is an alias for `none`; it means that this rule only uses `pattern-regex` and will operate on text files.

**Comment by minusworld** [comment]
> @minusworld I'm thrilled about this PR! Couple questions:
> 
> 1. Is there any benefit to restricting regex to string literals? Curious if it helps with either accuracy and/or speed (reducing the search space, but increasing the parsing?)
> 2. Are there any file extensions that can be used to reduce the search space? I noticed while looking at other "generic" rules we haven't made a habit of it, and curious here if it would help  or be possible.

(1) In my review of exposed secrets, I commonly found them in:
- Shell scripts of many varieties (.sh, .bash, .zsh, etc.)
- Code
- Logs

So, it's hard to say that they'll always be string literals. They could be hiding anywhere.

(2) Once merged, I'd like to run the rules at scale without any file restrictions. This will help us determine whether there are common files we can restrict to. However, I think for the largest search space, we won't want to restrict these. As mentioned above, secrets can be hiding in lots of places...

That said, this reminds me that our [rule benchmarks](https://github.com/returntocorp/semgrep-rules/actions/workflows/rule-benchmarks.yaml) don't run `generic` or `regex` rules... 🤔

**Pull request merged**